oss_src/unity/python/sframe/data_structures/sarray.py

```diff
@@ -1873,6 +1873,36 @@ def sample(self, fraction, seed=None):
         with cython_context():
             return SArray(_proxy=self.__proxy__.sample(fraction, seed))
 
+    def hash(self, seed=0):
+        """
+        Returns an SArray with a hash of each element. seed can be used
+        to change the hash function to allow this method to be used for
+        random number generation.
+
+        Parameters
+        ----------
+        seed : int
+            Defaults to 0. Can be changed to different values to get 
+            different hash results.
+
+        Returns
+        -------
+        out : SArray
+            An integer SArray with a hash value for each element. Identical
+            elements are hashed to the same value
+        """
+        with cython_context():
+            return SArray(_proxy=self.__proxy__.hash(seed))
+
+    @classmethod
+    def random_integers(self, size, seed=None):
+        """
+        Returns an SArray with random integer values.
+        """
+        if seed is None:
+            seed = time.time()
```

Comment:
This has resolution up to a second, so calls within the same second produce the same values.  Use ` seed = hash("%0.20f" % time.time())`.